src/phpDocumentor/Transformer/Template/Factory.php

```diff
@@ -173,6 +205,20 @@ private function resolve(Transformer $transformer, string $nameOrPath) : Filesys
         throw new TemplateNotFound($nameOrPath);
     }
 
+    private function getTemplatesDirectory() : Filesystem
+    {
+        $dsnString = $this->getTemplatesPath();
+        try {
+            $filesystem = $this->flySystemFactory->create(Dsn::createFromString($dsnString));
```

Comment:
Not for now, but it might be better to instantiate the filesystems in the config loading phase and inject them here. This will remove the need for passing DSNs around